org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/preferences/Preferences.java

```diff
@@ -443,7 +444,7 @@ private Preferences setIncompleteClasspathSeverity(Severity severity) {
 
 	public Preferences setImportOrder(List<String> importOrder) {
 		this.importOrder = (importOrder == null || importOrder.size() == 0) ? JAVA_IMPORT_ORDER_DEFAULT : importOrder;
-		IEclipsePreferences pref = JavaLanguageServerPlugin.getPreferencesManager().getEclipsePreferences();
+		IEclipsePreferences pref = InstanceScope.INSTANCE.getNode(JavaLanguageServerPlugin.PLUGIN_ID);
```

Comment:
I don't like the IEclipsePreferences coupling here. Can we move that part to an IPreferencesChangeListener instance instead? The idea being that if oldPreferences.importOrder != newPreferences.importOrder, that'd trigger that IEclipsePreferences change

Reply:
EclipsePreferences already works thhat way. See EclipsePreferences.put(String, String)

Reply:
no, I'm talking about changing IEclipsePreferences in a org.eclipse.jdt.ls.core.internal.preferences.IPreferencesChangeListener instance, to decouple our Preferences class from IEclipsePreferences.

But that's an improvement we can tackle later, for now, I'll settle for having the tests fixed, and that version of the fix is way better than 1st one you proposed.